src/c_kzg_4844.c

```diff
@@ -1796,29 +1723,17 @@ C_KZG_RET load_trusted_setup(
         blst_p2_from_affine(&out->g2_values[i], &g2_affine);
     }
 
-    /* It's the smallest power of 2 >= n1 */
-    unsigned int max_scale = 0;
-    while ((1ULL << max_scale) < n1)
-        max_scale++;
-
     /* Initialize the KZGSettings struct */
-    ret = c_kzg_malloc((void **)&out->fs, sizeof(FFTSettings));
-    if (ret != C_KZG_OK) goto out_error;
-    ret = new_fft_settings(out->fs, max_scale);
-    if (ret != C_KZG_OK) goto out_error;
-    ret = fft_g1(out->g1_values, g1_projective, true, n1, out->fs);
+    ret = new_roots_of_unity(out->roots_of_unity, max_scale);
     if (ret != C_KZG_OK) goto out_error;
     ret = bit_reversal_permutation(out->g1_values, sizeof(g1_t), n1);
     if (ret != C_KZG_OK) goto out_error;
 
     goto out_success;
 
 out_error:
-    c_kzg_free(out->fs);
-    c_kzg_free(out->g1_values);
-    c_kzg_free(out->g2_values);
+    free_trusted_setup(out);
```

Comment:
This is slightly confusing because usually `free_*()` helpers also free the input pointer but the public func `free_trusted_setup()` doesn't. This is the correct behavior because this function should not free `out`, but it might confuse the reader.

No strong opinion whether explicit `c_kzg_free()` are better, but thought I should mention it.

Reply:
Hmm good point. Worthy of a comment?

Reply:
Yes, a comment could do it.

Reply:
Made a comment. Feel free to recommend something else.

Reply:
Looks good. Thanks.